src/Components/Admin/AssetModelViewer/assetModelViewer.js

```diff
@@ -0,0 +1,153 @@
+import React, { useEffect, useState } from "react";
+import "@google/model-viewer";
+import { makeStyles } from "@material-ui/core/styles";
+import Drawer from "@material-ui/core/Drawer";
+import CssBaseline from "@material-ui/core/CssBaseline";
+import Typography from "@material-ui/core/Typography";
+import Divider from "@material-ui/core/Divider";
+import Button from "@material-ui/core/Button";
+import TextField from "@material-ui/core/TextField";
+import { getAsset, editAsset } from "../../../lib/assetEndpoints";
+import { useErrorHandler } from "react-error-boundary";
+import { jsonMock } from "./mockData.js";
+import SaveIcon from "@material-ui/icons/Save";
+
+const drawerWidth = 17;
+
+const useStyles = makeStyles((theme) => ({
+    root: {
+        display: "flex",
+    },
+    drawer: {
+        width: drawerWidth + "vw",
+        flexShrink: 0,
+    },
+    drawerPaper: {
+        width: drawerWidth + "vw",
+    },
+    // necessary for content to be below navbar
+    toolbar: theme.mixins.toolbar,
+    content: {
+        flexGrow: 1,
+        backgroundColor: theme.palette.background.default,
+        padding: theme.spacing(3),
+    },
+    textfields: {
+        "& .MuiTextField-root": {
+            margin: theme.spacing(2),
+            width: "15vw",
+        },
+    },
+}));
+
+export default function AssetModelViewer({
+    match: {
+        params: { assetId },
+    },
+}) {
+    const classes = useStyles();
+    const handleError = useErrorHandler();
+    const [asset, setAsset] = useState({});
+    const [name, setName] = useState("");
+    const [objectType, setObjectType] = useState("");
+    const [buttonText, setButtonText] = useState("Save");
+    const [modelGLB, setModelGLB] = useState("");
+
+    useEffect(() => {
+        const getObjectAsset = async () => {
+            const data = await getAsset(assetId, handleError);
+            setName(data.name);
+            setObjectType(data.obj_type);
+            // TODO: Replace with asset's s3_link once it contains a legit link
+            setModelGLB(jsonMock.linksGLB[0]);
+            setAsset(data);
+        };
+
+        if (assetId) {
+            getObjectAsset();
+        }
+    }, [assetId, handleError]);
+
+    const handleNameChange = (event) => {
+        setName(event.target.value);
+    };
+
+    const handleObjectTypeChange = (event) => {
+        setObjectType(event.target.value);
+    };
+
+    async function updateAsset(event) {
+        event.preventDefault();
+        setButtonText("Saved!");
+        setTimeout(() => {
+            setButtonText("Save");
+        }, 2000);
+        editAsset(assetId, name, objectType, asset.s3_key, handleError);
+    }
```

Comment:
@ahmedhamodi is the way that we did `setTimeout` best practice? I almost never use `setTimeout` for frontend work because it could be blocking but idk.

Reply:
I've never used setTimeout before. I also don't think it's needed here. Was there any reason you added it Vivian?

Reply:
I added it so that when the user clicks on the save button, it changes to "saved!" for 2 seconds before returning back to the original "save" state. Wasn't sure if there was a better way to do this without setTimeout

Reply:
We can keep this in for now but I remember someone telling me using setTimeout on the frontend is not a good idea. However, don't want this to block you from moving forward so we can keep this in until the designers have a better idea